history.go

```diff
@@ -54,11 +68,15 @@ func (h *History) UpdateHistory(policies Policies, m *MerkleTree, nextEp Epoch)
 		}
 		h.loadedEpochs = append(h.loadedEpochs[:0], h.loadedEpochs[n:]...)
 	}
-	h.snapshots[nextEp] = nextStr
-	h.loadedEpochs = append(h.loadedEpochs, nextEp)
+
+	h.updateInternal(policies, nextEp)
 	return nil
 }
 
+func (h *History) Set(key string, value []byte) error {
```

Comment:
What do you mean "the server level"? Is it our libconiks-server library? 
I think it's fine to have these wrappers in this library. We can think these wrappers as APIs of the library, and in the libconiks-server library, we should have something like `directory.LookUp` and `directory.Set()`. Is that what you mean?